File: 3666_Minimum_Operations_to_Equalize_Binary_String/solution.c
int minOperations(char* s, int k) {
    int n = strlen(s), z = 0;
    for (int i = 0; i < n; i++) z += ~s[i] & 1;

    if (z == 0) return 0;
    if (n == k) return z == n ? 1 : -1;

    int b = n - k;
    int o = fmax(ceil((double)z / k), ceil((double)(n - z) / b));
    o += ~o & 1;
    int e = fmax(ceil((double)z / k), ceil((double)z / b));
    e += e & 1;

    int r = INT_MAX;
    if ((k & 1) == (z & 1)) r = fmin(r, o);
    if ((~z & 1) == 1) r = fmin(r, e);
    return r == INT_MAX ? -1 : r;
}

File: 3666_Minimum_Operations_to_Equalize_Binary_String/solution.py
class Solution:
    def minOperations(self, s: str, k: int) -> int:
        n = len(s)
        z = sum(~ord(c) & 1 for c in s)

        if z == 0: return 0
        if n == k: return 1 if z == n else -1

        b = n - k
        o = max(math.ceil(z / k), math.ceil((n - z) / b))
        o += ~o & 1
        e = max(math.ceil(z / k), math.ceil(z / b))
        e += e & 1

        r = min(o if (k & 1) == (z & 1) else float("inf"), e if (~z & 1) == 1 else float("inf"),)

        return -1 if r == float("inf") else r